.bin/esyBuildCommand.js

```diff
@@ -0,0 +1,323 @@
+const childProcess = require('child_process');
+const path = require('path');
+
+const ESY_SANDBOX_REF = '$(ESY__SANDBOX)';
+
+function installDir(pkgName, ...args) {
+  return path.join(
+    '$(ESY__SANDBOX)', '_install', 'node_modules', pkgName, ...args);
+}
+
+function buildDir(pkgName, ...args) {
+  return path.join(
+    '$(ESY__SANDBOX)', '_build', 'node_modules', pkgName, ...args);
+}
+
+function envToEnvList(env) {
+  return env.envVars.map(env => ({
+    name: env.name,
+    value: env.normalizedVal
+  }));
+}
+
+function buildCommand(curDir, env, args) {
+
+  // first group is a shared (built-in) env
+  env = env.slice();
+  let builtIns = env.shift();
+
+  let envMap = env.reduce((envMap, pkg) => {
+    envMap.set(pkg.packageJson.name, pkg);
+    return envMap;
+  }, new Map());
+
+  let rules = [];
+
+  let prelude = [
+    {
+      type: 'define',
+      name: 'ESY__SANDBOX',
+      value: '$(PWD)',
+      assignment: '?=',
+    },
+  ];
+
+  env.forEach(pkg => {
+    let info = pkg.packageJson;
+    let dependencies = collectTransitiveDependencies(envMap, info);
+
+    let findlibPath = dependencies
+      .map(dep => installDir(dep, 'lib'))
+      .join(':');
+
+    prelude.push({
+      type: 'define',
+      multiline: true,
+      name: `${info.name}__FINDLIB_CONF`,
+      value: `
+path = "$(shell ocamlfind printconf path):${findlibPath}"
```

Comment:
One challenge is that this makes `ocamlfind` a mandatory dependency of `esy`. It's not bad if it's a requirement of the example build system, but I feel like we shouldn't need it for `esy` itself.

(Even then, it would be nice to not require `ocamlfind` for the example build system either - we can make a mock `ocamlfind` script).